server/server.go

```diff
@@ -948,27 +950,43 @@ func (s *Server) generateCRIEvent(ctx context.Context, container *oci.Container,
 
 	sandboxStatuses, err := s.getSandboxStatuses(ctx, s.GetSandbox(container.Sandbox()).ID())
 
-	if IsNotFound(err) {
+	if isNotFound(err) {
 		return nil
 	}
 
 	if err != nil {
-		return fmt.Errorf("generateCRIEvent: failed to get sandbox statuses of the pod %s due to err %v", sandboxStatuses.Metadata.Uid, err)
+		log.Errorf(ctx, "generateCRIEvent: event type: %s, failed to get sandbox statuses of the pod %s: %w", eventType, sandboxStatuses.Metadata.Uid, err)
+		return err
 	}
 
 	containerStatuses, err := s.getContainerStatuses(ctx, sandboxStatuses.Metadata.Uid)
 
 	if err != nil {
-		return fmt.Errorf("generateCRIEvent: failed to get container statuses of the pod %s due to err %v", sandboxStatuses.Metadata.Uid, err)
+		log.Errorf(ctx, "generateCRIEvent: event type: %s, failed to get container statuses of the pod %s: %w", eventType, sandboxStatuses.Metadata.Uid, err)
+		return err
 	}
 
 	select {
 	case s.ContainerEventsChan <- types.ContainerEventResponse{ContainerId: container.ID(), ContainerEventType: eventType, CreatedAt: time.Now().UnixNano(),
 		PodSandboxStatus: sandboxStatuses, ContainersStatuses: containerStatuses}:
 		log.Debugf(ctx, "Container event %s generated for %s", eventType, container.ID())
 	default:
-		return fmt.Errorf("generateCRIEvent: failed to generate event %s for container %s", eventType, container.ID())
+		err := fmt.Errorf("generateCRIEvent: failed to generate event %s for container %s", eventType, container.ID())
+		log.Errorf(ctx, "%w", err)
```

Comment:
Only log if `err != nil`, otherwise we will throw an `<nil>` log message.

Reply:
This `err` can never be a `nil` right?

Reply:
Ugh, my fault :+1: